arches/install/arches-admin

```diff
@@ -0,0 +1,246 @@
+)
+
+subparsers = parser.add_subparsers(title='available commands', dest='command')
+subparsers.required = True
+
+parser_startproject = subparsers.add_parser(
+    'startproject',
+    help="Create the scaffolding for a new Arches project",
+)
+parser_startproject.add_argument(
+    'name',
+    type=str,
+    help='name of your new project'
+)
+parser_startproject.add_argument(
+    '-d', '--directory',
+    help='destination directory of your new project',
+    dest='directory',
+    default=None,
+)
+parser_startproject.add_argument(
+    '-t', '--template',
+    help="The path or URL to load the template from.",
+    type=str,
+    default=os.path.join(os.path.dirname(arches.__file__), 'install', 'arches-templates')
+)
+parser_startproject.add_argument(
+    '-e', '--extension',
+    dest='extensions',
+    help='The file extension(s) to render (default: py).',
+    type=str,
+    default=['py', 'txt', 'html', 'js', 'css', 'log', 'json', 'gitignore']
+)
+parser_startproject.add_argument(
+    '-n', '--name',
+    dest='files',
+    help='name of your new arches project',
+    type=str,
+    default='',
+)
+
+parser_startapp = subparsers.add_parser(
+    'startapp',
+    help="Create the scaffolding for a new Arches application",
+)
+parser_startapp.add_argument(
+    'name',
+    type=str,
+    help='name of your new application'
+)
+parser_startapp.add_argument(
+    '-d', '--directory',
+    help='destination directory of your new project',
+)
+parser_startapp.add_argument(
+    '-t', '--template',
+    help="Provides the path to a directory with a custom app template file, or a path to an uncompressed archive, a compressed archive, or a URL containing the app template files.",
+    type=str,
+    default=os.path.join(os.path.dirname(arches.__file__), 'install', 'arches-app-templates')
+)
+parser_startapp.add_argument(
+    '-e', '--extension',
+    dest='extensions',
+    help='Specifies which file extensions in the app template should be rendered with the template engine. Defaults to py.',
+    type=str,
+    default='py'
+)
+parser_startapp.add_argument(
+    '-n', '--name',
+    dest='files',
+    help='Specifies which files in the app template (in addition to those matching --extension) should be rendered with the template engine.',
+    type=str,
+    default='',
+)
+
+class ArchesProjectCommand(TemplateCommand):
+    help = (
+        "Creates a Django project directory structure for the given "
+        "project or application name in the current directory or "
+        "optionally in the given directory."
+    )
+    missing_args_message = "You must provide a valid name."
+
+    def handle(self, options):
+        project_name, target = options.pop('name'), options.pop('directory')
+
+        # Check that the project_name cannot be imported.
+        try:
+            self.validate_name(project_name, "project")
+            import_module(project_name)
```

Comment:
Instead of depending on an import, you can just check that [find_spec()](https://docs.python.org/3/library/importlib.html?highlight=find_spec#importlib.util.find_spec) returns not None. I don't think Arches is to blame if a user has untrusted scripts or spammy pypi modules hanging around in their environment, but it's still nicer/more performant not to execute arbitrary code.

Interestingly, importlib will execute the parent module if you provide a dotted path, so you could be super fastidious and check for that first and raise an error.

Here I altered `arches.__init__.py` to show when it's being executed:
```py
>>> from importlib.util import find_spec
>>> find_spec('arches')
ModuleSpec(name='arches', loader=<_frozen_importlib_external.SourceFileLoader object at 0x102f33f70>, origin='/Users/jwalls/prj/arches/arches/__init__.py', submodule_search_locations=['/Users/jwalls/prj/arches/arches'])
>>> find_spec('arches.tests')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/util.py", line 94, in find_spec
    parent = __import__(parent_name, fromlist=['__path__'])
  File "/Users/jwalls/prj/arches/arches/__init__.py", line 4, in <module>
    1/0
ZeroDivisionError: division by zero
```